app/models/driver.rb

```diff
@@ -0,0 +1,64 @@
+class Driver < ApplicationRecord
+  has_many :trips
+  validates :name, presence: true
+  validates :vin, presence: true
+
+  def earnings
+    earnings = 0
+    self.trips.each do |trip|
+      next if trip.cost.nil?
+      earnings += trip.cost
+    end
+    return earnings
+  end
+
+  def average_rating
+    count = 0
+    stars = 0
+
+    return 0 if trips.empty? || (trips.length == 1 && trips.any? {|trip| trip.rating.nil? })
+
+    self.trips.each do |trip|
+      next if trip.rating.nil?
+      count +=1
+      stars += trip.rating
+    end
+    return stars/count
+  end
+
+  def last_trip
+    drivers_trips = self.trips.order(date: :desc)
+    if drivers_trips.length == 0
+      nil
+    end
+    return drivers_trips.first
+  end
+
+  def self.the_driver
+    drivers = Driver.where(disabled: false)
+    longest_driver_not_driving = drivers.first
+    drivers.each do |driver|
```

Comment:
I'm not sure about the name of this method - if I hadn't seen this file and I saw a call to `Driver.the_driver`, I would not know what that method is doing. Something like `select_driver` might be more mnemonic.